src/components/Video.tsx

```diff
@@ -0,0 +1,63 @@
+import React from 'react';
+import DownloadButton from './DownloadButton';
+import FavoriteButton from './FavoriteButton';
+import Label from './Label'
+import { get } from 'idb-keyval';
+import { videoStore } from '../index'
+import { Resource, videoOrArticle } from '../store/resourcesSlice'
+import { makeStyles, createStyles } from "@material-ui/core/styles"
+
+type VideoProps = {
+  resId: number,
+  videoDetails: Resource,
+}
+
+const styles = makeStyles(() => createStyles({
+  labelList: {
+    display: "flex",
+    flexDirection: "row",
+
+  }
+}))
+
+const Video = ({ resId, videoDetails }: VideoProps) => {
+
+  const classes = styles();
+
+  function getId(url) {
+    const regExp = /^.*(youtu.be\/|v\/|u\/\w\/|embed\/|watch\?v=|&v=)([^#&?]*).*/;
+    const match = url.match(regExp);
+    return (match && match[2].length === 11) ? match[2] : null;
+  }
+
+  let videoUrl;
+
+  if (videoDetails.isCached) {
+    get(resId, videoStore).then(videoBlob => {
```

Comment:
We should have a `.catch` here in case loading the cached video fails. This might be due to an IndexedDB error (in which case we could default to downloading the video again), or because for some reason the resource is no longer cached even though it's still marked as cached in Redux (in which case we should update Redux)

Reply:
That makes sense! I'll take a look into this!